lib/validators/check-string-param.js

```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+const constants = require('../constants.js')
+
+module.exports = function checkStringParam(param, name) {
+  if (!param || typeof param !== 'string') {
+    throw new TypeError(`${name} is undefined or not passed as a String`)
+  }
+  if (param.length > constants.MAX_INPUT_LENGTH) {
+    throw new TypeError(
+      `${name} cannot be longer than ${constants.MAX_INPUT_LENGTH} chars`
+    )
+  }
```

Comment:
Client side enforced limits should reflect our document ingestion API limits: https://docs.logdna.com/docs/ingestion#service-limits

Reply:
No idea if this is up to do or not, or whether the ingester even enforces these limits. We should definitely figure this out and try to be consistent across our clients and api.

Reply:
yeah, this "restriction" is carry-over from the previous version, and I honestly don't know where it came from.  I kinda think this is a bit over-engineered since the limits we have in that link are REALLY big for things like app names.

I like the idea of tightening this up, but I think we can iterate that in a future commit so that those changes stand on their own.  Since this will require digging into the ingester (like you said, are the values accurate?), we shouldn't block this PR now.